src/lib/table/cell.ts

```diff
@@ -71,6 +71,12 @@ export class MatFooterCellDef extends CdkFooterCellDef {
 export class MatColumnDef extends CdkColumnDef {
   /** Unique name for this column. */
   @Input('matColumnDef') name: string;
+
+  /** Whether this column should be sticky positioned on the left of the right */
+  @Input() stickyLeft: boolean;
```

Comment:
The vast, vast majority of cases are doing to want to stick to the start(left) side. Optimizing for the common case, I think we should just make the start/left position `sticky` and the other one `stickyEnd`. 

We should talk separately about whether this should be something like `cdkColumnSticky`

Reply:
SGTM, I agree the vast majority wants to just use `sticky` and mean it to be the start.